Chat settings UI

dlqqq opened the pull request:
# Description

https://user-images.githubusercontent.com/44106031/236072832-4b8d9055-0f7e-4469-8fdc-3a0474d5a3a9.mov

Implement basic chat settings UI that allows users to switch language models, embedding models, and add API keys. Other changes:

- Model config is now created automatically on initialization if one doesn't exist
- Prefer upper-case environment variables in the config file

# Screenshots

<img width="405" alt="Screen Shot 2023-05-03 at 11 50 14 AM" src="https://user-images.githubusercontent.com/44106031/236015284-a77c97f7-cbe2-4502-9e4b-37461f236949.png">

This UI also handles loading and error states for the initial fetch. The loading state is an animated spinner, though this isn't captured well via screenshot.

<img width="555" alt="Screen Shot 2023-05-02 at 2 34 21 PM" src="https://user-images.githubusercontent.com/44106031/236015483-56947cae-6975-46c1-b32b-566743dc3569.png">

<img width="683" alt="Screen Shot 2023-05-02 at 2 38 23 PM" src="https://user-images.githubusercontent.com/44106031/236015369-f8e22470-ba48-4950-9521-9431b494c579.png">

3coins commented:
@dlqqq 
Thanks for working on this. It works great, makes it really simple to switch models 🚀 . 

Here are a few observations:
1. I started with no config, and did not see any error or initial screen prompting me to add these settings before using the chat ui. As I used the chat, I got this error.
    <img width="457" alt="Screen Shot 2023-05-03 at 12 03 39 PM" src="https://user-images.githubusercontent.com/289369/236020299-7a8a33e5-110c-40b3-a086-cd45ef6de7c8.png">
    We should rather start with the settings panel, if there is no config entered, with a message at the top that these values are required for chat usage.

2. The save link does not have any indication when you click to save the values, we should show a message and return back to chat, once the values are saved. **Minor**: We could use a button rather than a link for the save action, and align it to the right. 

3. The list component for the model selection is too long, can we limit the height of the popup?

4. Minor: Use a different settings icon, this one conflicts with the gear icon in the right panel.

dlqqq commented:
@3coins Thank you for the poignant feedback! I addressed all your points with the exception of the settings icon; I think we'll want some input from a UXD before changing it.

3coins commented:
@dlqqq 
Thanks for making all the changes, the demo looks great 👍. 
I corrected the hint string for the chat input, and made a slight change in the welcome message. Rather than asking users to find the settings icon, I added a button in the message to take them there. Please feel free to update if this doesn't look ok and merge.
<img width="361" alt="Screenshot 2023-05-03 at 7 17 17 PM" src="https://user-images.githubusercontent.com/289369/236096917-c399639f-63b0-4c2a-b418-c828982d1d84.png">

dlqqq commented:
That looks great, thank you!
